load.php

```diff
@@ -253,9 +263,101 @@ function perflab_load_active_and_valid_modules() {
+	global $wp_filesystem;
+
+	// Bail if disabled via constant.
+	if ( defined( 'PERFLAB_DISABLE_OBJECT_CACHE_DROPIN' ) && PERFLAB_DISABLE_OBJECT_CACHE_DROPIN ) {
+		return;
+	}
+
+	// Bail if already placed.
+	if ( PERFLAB_OBJECT_CACHE_DROPIN_VERSION ) {
+		return;
+	}
+
+	// Bail if already attempted before timeout has been completed.
+	// This is present in case placing the file fails for some reason, to avoid
+	// excessively retrying to place it on every request.
+	$timeout = get_transient( 'perflab_set_object_cache_dropin' );
+	if ( false !== $timeout ) {
+		return;
+	}
+
+	if ( $wp_filesystem || WP_Filesystem() ) {
+		// If there is an actual object-cache.php file, rename it.
+		// The Performance Lab object-cache.php will still load it, so the
+		// behavior does not change.
+		if ( $wp_filesystem->exists( WP_CONTENT_DIR . '/object-cache.php' ) ) {
+			$wp_filesystem->move( WP_CONTENT_DIR . '/object-cache.php', WP_CONTENT_DIR . '/object-cache-plst-orig.php' );
+		}
+
+		$wp_filesystem->copy( PERFLAB_PLUGIN_DIR_PATH . 'server-timing/object-cache.copy.php', WP_CONTENT_DIR . '/object-cache.php' );
+	}
+
+	// Set timeout of 1 hour before retrying again (only in case of failure).
+	set_transient( 'perflab_set_object_cache_dropin', true, HOUR_IN_SECONDS );
+}
+add_action( 'admin_init', 'perflab_maybe_set_object_cache_dropin' );
+
+/**
+ * Removes the Performance Lab's object cache drop-in from the drop-ins folder.
+ *
+ * This function should be run on plugin deactivation. If there was another original
+ * object-cache.php drop-in file (renamed in `perflab_maybe_set_object_cache_dropin()`
+ * to object-cache-plst-orig.php), it will be restored.
+ *
+ * This function will short-circuit if the constant
+ * 'PERFLAB_DISABLE_OBJECT_CACHE_DROPIN' is set as true.
+ *
+ * @since n.e.x.t
+ *
+ * @global WP_Filesystem_Base $wp_filesystem WordPress filesystem subclass.
+ */
+function perflab_maybe_remove_object_cache_dropin() {
+	global $wp_filesystem;
+
+	// Bail if disabled via constant.
+	if ( defined( 'PERFLAB_DISABLE_OBJECT_CACHE_DROPIN' ) && PERFLAB_DISABLE_OBJECT_CACHE_DROPIN ) {
+		return;
+	}
+
+	// Bail if custom drop-in not present anyway.
+	if ( ! PERFLAB_OBJECT_CACHE_DROPIN_VERSION ) {
+		return;
+	}
+
+	if ( $wp_filesystem || WP_Filesystem() ) {
+		// If there is an actual object-cache.php file, restore it
+		// and override the Performance Lab file.
+		// Otherwise just delete the Performance Lab file.
+		if ( $wp_filesystem->exists( WP_CONTENT_DIR . '/object-cache-plst-orig.php' ) ) {
+			$wp_filesystem->move( WP_CONTENT_DIR . '/object-cache-plst-orig.php', WP_CONTENT_DIR . '/object-cache.php', true );
+		} else {
+			$wp_filesystem->delete( WP_CONTENT_DIR . '/object-cache.php' );
+		}
+	}
+
+	// Delete transient for drop-in check in case the plugin is reactivated shortly after.
+	delete_transient( 'perflab_set_object_cache_dropin' );
+}
+register_deactivation_hook( __FILE__, 'perflab_maybe_remove_object_cache_dropin' );
```

Comment:
The deactivate hook calls before the uninstall hook, so I don't think we need to add in `uninstall.php`.

Reply:
@eugene-manuilov Agree with @mukeshpanchal27, the file should no longer be used as soon as the plugin is no longer active. Otherwise it'll keep loading the Server-Timing code from the plugin even when the plugin is not active.

Reply:
Ok, got it.